tests/integration/test_dictionaries_update_and_reload/test.py

```diff
@@ -37,7 +37,7 @@ def get_status(dictionary_name):
     ).rstrip("\n")
 
 
-def get_status_retry(dictionary_name, expect, retry_count=10, sleep_time=0.5):
+def get_status_retry(dictionary_name, expect, retry_count=50, sleep_time=0.5):
```

Comment:
`test_reload_after_fail_by_timer` will not be fixed by this, no matter how many retries do you have, the problem is not them anyway, the problem is that you are expecting `FAILED` after `get_status_retry`, but this is not guarnateed - https://s3.amazonaws.com/clickhouse-test-reports/66311/316dd790bdb5a8cb4aef11c356f14700f3b9169e/integration_tests__asan__old_analyzer__[3_6].html

If you will look properly at assert you will see.

```
        # on sanitizers builds it can return 'FAILED_AND_RELOADING' which is not quite right
        # add retry for these builds
        if (
            instance.is_built_with_sanitizer()
            and get_status("no_file_2") == "FAILED_AND_RELOADING"
        ):
            get_status_retry("no_file_2", expect="FAILED") # here it already gets `FAILED` but it got `FAILED_AND_RELOADING` after anyway, and this is possible
    
>       assert get_status("no_file_2") == "FAILED"
E       AssertionError: assert 'FAILED_AND_RELOADING' == 'FAILED'
E         - FAILED
E         + FAILED_AND_RELOADING
```

Also I will wrote this one more time - **you can get `FAILED_AND_RELOADING` even for release build**

Plus no need to make test slower with this retries when it can be done without

**So I think this `get_status_retry` should gone**

Reply:
1.  > ```# here it already gets `FAILED` but it got `FAILED_AND_RELOADING` after anyway, and this is possible```

`get_status_retry` does not assert value, it only wait for `FAIL` for only 5 seconds and continues even if status is `FAILED_AND_RELOADING`. That's why fail is on the next line.
2. I didn't see any fails because of `FAILED_AND_RELOADING` in release builds
3. Solution to expect both statuses is incorrect, we expect only `FAIL` status

Reply:
>2. I didn't see any fails because of FAILED_AND_RELOADING in release builds

It doesn't says that it is impossible

>Solution to expect both statuses is incorrect, we expect only FAIL status

Can you please explain why? The test expecting that the loading will fail, if the status is FAILED_AND_RELOADING, it means that the load had been failed already and this is OK for the test

>get_status_retry does not assert value, it only wait for FAIL for only 5 seconds

Correct

> and continues even if status is FAILED_AND_RELOADING. That's why fail is on the next line.

Wrong:

```
def get_status_retry(dictionary_name, expect, retry_count=10, sleep_time=0.5):
    for _ in range(retry_count):
        res = get_status(dictionary_name)
        if res == expect:
            return res
        time.sleep(sleep_time)

    raise Exception(f'Expected result "{expect}" did not occur')
```

It will raise an exception if the status will not match unless I forgot how to read python.

Reply:
I didn't expect that the status could change from `FAIL` to `FAILED_AND_RELOADING`, so expecting both statuses seems correct